pkg/api/types.go

```diff
@@ -584,6 +584,8 @@ type DownwardAPIVolumeFile struct {
 	Path string `json:"path"`
 	// Required: Selects a field of the pod: only annotations, labels, name and  namespace are supported.
 	FieldRef ObjectFieldSelector `json:"fieldRef"`
+	// Required: Selects a field of the pod; only name and namespace are supported.
+	ConfigDataRef *ConfigDataSelector `json:"configDataRef"`
```

Comment:
@pmorie I added a placeholder type for selecting fields but I realized that FieldRef has hard-coded ``Pod`` into it. We'll be better off refactoring ``ObjectFieldSelector``. The reason I modified the api/types.go instead of experimental is that I couldn't extend a struct in another pkg. I'll copy the whole struct to experimental next time.
@smarterclayton agreed. FieldRef refactoring is much better than duplication.